odoo/tests/common.py

```diff
@@ -1378,21 +1378,43 @@ def replacer(m):
         return replacer
 
 
+class Opener(requests.Session):
+    """
+    Flushes and clears the current transaction when starting a request.
+
+    This is likely necessary when we make a request to the server, as the
+    request is made with a test cursor, which uses a different cache than this
+    transaction.
+    """
+    def __init__(self, cr: BaseCursor):
+        super().__init__()
+        self.cr = cr
+
+    def request(self, *args, **kwargs):
+        self.cr.flush(); self.cr.clear()
```

Comment:
Maybe add a comment on why we need this:
* `cr.flush()` processes all pending computations and updates, so that the request (using a test cursor) sees the database as a new transaction would
* `cr.clear()` clears the cache to make it consistent *after* the request, as the request may change the database but the changes are not propagated to the current transaction's environments

Reply:
any good reason for a one line ?

Reply:
You are right, both statements are necessary, and for distinct reasons.
It should be two lines.
Same below.

Reply:
I refuse paying taxes on newlines!

More seriously, it looks / feels more atomic that way, we're trying to do one thing even if we need two calls to.